tests/test_buffer.py

```diff
@@ -239,24 +239,31 @@ def check_identical(sarr, ndarr):
         ndarrT = ndarr.transpose()
 
         sarr = modmesh.SimpleArrayFloat64(array=ndarr)
-        noarr = sarr.transpose()
+        sarr2 = sarr.transpose()
```

Comment:
Add a check to assert that the internal memory buffer of `sarr` and `sarr2` are the same.